__tests__/schema/taxonomy-term/set-name-and-description.ts

```diff
@@ -0,0 +1,158 @@
+/**
+ * This file is part of Serlo.org API
+ *
+ * Copyright (c) 2020-2022 Serlo Education e.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @copyright Copyright (c) 2020-2022 Serlo Education e.V.
+ * @license   http://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/serlo-org/api.serlo.org for the canonical source repository
+ */
+import { gql } from 'apollo-server'
+
+import {
+  taxonomyTermCurriculumTopic,
+  user as baseUser,
+} from '../../../__fixtures__'
+import { Client, given, nextUuid } from '../../__utils__'
+
+describe('TaxonomyTermSetNameAndDescriptionMutation', () => {
+  const user = { ...baseUser, roles: ['de_architect'] }
+
+  const input = {
+    description: 'a description',
+    name: 'a name',
+    id: taxonomyTermCurriculumTopic.id,
+  }
+
+  const mutation = new Client({ userId: user.id })
+    .prepareQuery({
+      query: gql`
+        mutation set($input: TaxonomyTermSetNameAndDescriptionInput!) {
+          taxonomyTerm {
+            setNameAndDescription(input: $input) {
+              success
+            }
+          }
+        }
+      `,
+    })
+    .withVariables({ input })
+
+  beforeEach(() => {
+    given('UuidQuery').for(user, taxonomyTermCurriculumTopic)
+  })
+
+  test('returns "{ success: true }" when mutation could be successfully executed', async () => {
+    given('TaxonomyTermSetNameAndDescriptionMutation')
+      .withPayload({
+        ...input,
+        userId: user.id,
+      })
+      .returns({ success: true })
+
+    await mutation.shouldReturnData({
+      taxonomyTerm: { setNameAndDescription: { success: true } },
+    })
+  })
+
+  test('fails when user is not authenticated', async () => {
+    await mutation
+      .forUnauthenticatedUser()
+      .shouldFailWithError('UNAUTHENTICATED')
+  })
+
+  test('fails when user does not have role "architect"', async () => {
+    const loginUser = { ...user, id: nextUuid(user.id), roles: ['login'] }
+
+    given('UuidQuery').for(loginUser)
+
+    await mutation
+      .forClient(new Client({ userId: loginUser.id }))
+      .withVariables({ input })
+      .shouldFailWithError('FORBIDDEN')
+  })
+
+  test('fails when `name` is empty', async () => {
+    await mutation
+      .withVariables({
+        input: {
+          ...input,
+          name: '',
+        },
+      })
```

Comment:
```suggestion
      .withVariables({ input: { ...input, name: '' } })
```

i like when it is short